pyuvm/s18_pyuvm_reg_block.py

```diff
@@ -0,0 +1,19 @@
+from pyuvm import uvm_object
+#import vsc
+
+# 18.1.1 Class declaration
+class pyuvm_reg_block(uvm_object):
```

Comment:
Would it be possible to extend all classes in this change from `uvm_report_object` so that logging functionality is available via `self.logger.<info|warn|error|debug>`

@raysalemi , please correct me if this is not the case

Reply:
We need to follow the spec.  What does the spec say?

Reply:
the uvm report object is coming from uvm object. The standard uses uvm object directly perhaps the uvm object could have a reference to the report logger. Frankly if we have to follow the spec not sure if using the report object would make sense

Reply:
The user can use pyuvm logging without the object being a descendent of uvm_report_object. We do that all the time in sequences.

Reply:
ah ok. I've only seen logging using `uvm_root().logger.info`. Its good that its already available.

Reply:
Yes. That is how to do it. 

But, again, we must follow IEEE 1800.2 when it comes to inheritance.

Reply:
Agreed.